mne_connectivity/spectral/time.py

```diff
@@ -143,143 +281,150 @@ def spectral_connectivity_time(data, names=None, method='coh', indices=None,
-                           mode, sfreq, freqs, n_cycles, mt_bandwidth=None,
-                           decim=1, kw_cwt={}, kw_mt={}, n_jobs=1,
-                           verbose=False):
-    """EStimate time-resolved connectivity for one epoch.
+                           mode, sfreq, freqs, faverage, n_cycles,
+                           mt_bandwidth=None, decim=1, kw_cwt={}, kw_mt={},
+                           n_jobs=1, verbose=False):
+    """Estimate time-resolved connectivity for one epoch.
 
-    See spectral_connectivity_epoch."""
+    See spectral_connectivity_epochs."""
     n_pairs = len(source_idx)
 
-    # first compute time-frequency decomposition
-    collapse = None
     if mode == 'cwt_morlet':
         out = tfr_array_morlet(
             data, sfreq, freqs, n_cycles=n_cycles, output='complex',
             decim=decim, n_jobs=n_jobs, **kw_cwt)
+        out = np.expand_dims(out, axis=2)  # same dims with multitaper
     elif mode == 'multitaper':
-        # In case multiple values are provided for mt_bandwidth
-        # the MT decomposition is done separatedly for each
-        # Frequency center
-        if isinstance(mt_bandwidth, (list, tuple, np.ndarray)):
-            # Arrays freqs, n_cycles, mt_bandwidth should have the same size
-            assert len(freqs) == len(n_cycles) == len(mt_bandwidth)
-            out = []
-            for f_c, n_c, mt in zip(freqs, n_cycles, mt_bandwidth):
-                out += [tfr_array_multitaper(
-                    data, sfreq, [f_c], n_cycles=float(n_c), time_bandwidth=mt,
-                    output='complex', decim=decim, n_jobs=n_jobs, **kw_mt)]
-            out = np.stack(out, axis=3).squeeze()
-        elif isinstance(mt_bandwidth, (type(None), int, float)):
-            out = tfr_array_multitaper(
-                data, sfreq, freqs, n_cycles=n_cycles,
-                time_bandwidth=mt_bandwidth, output='complex', decim=decim,
-                n_jobs=n_jobs, **kw_mt)
-            collapse = True
-            if out.ndim == 5:  # newest MNE-Python
-                collapse = -3
-
-    # get the supported connectivity function
-    conn_func = {'coh': _coh, 'plv': _plv, 'sxy': _cs}[method]
-
-    # computes conn across trials
-    # TODO: This is wrong -- it averages in the complex domain (over tapers).
-    # What it *should* do is compute the conn for each taper, then average
-    # (see below).
-    if collapse is not None:
-        out = np.mean(out, axis=collapse)
-    this_conn = conn_func(out, kernel, foi_idx, source_idx, target_idx,
-                          n_jobs=n_jobs, verbose=verbose, total=n_pairs)
-    # This is where it should go, but the regression test fails...
-    # if collapse is not None:
-    #     this_conn = [c.mean(axis=collapse) for c in this_conn]
+        out = tfr_array_multitaper(
+            data, sfreq, freqs, n_cycles=n_cycles,
+            time_bandwidth=mt_bandwidth, output='complex', decim=decim,
+            n_jobs=n_jobs, **kw_mt)
```

Comment:
Is this change dependent on the discussion in #84 ?

Reply:
Sort of. The old code would just average over the complex signal. The code in this PR averages over the connectivity matrices. I think the discussion concluded with the notion that we should compute a weighted average over the CSD. Currently my local code computes a simple average of the CSD. To do it properly with the weighted average, we need to extract the weights for the tapers from the time-frequency module, which might require changes there.

From taking a quick look at the `tfr_array_multitaper` function it looks like the DPSS window eigenvalues are not used and therefore no weighting procedure is included. It might be necessary to modify that function such that it also returns the eigenvalues for the tapers. I will take a deeper look tomorrow and continue the discussion in #84.

I can eventually add the changes here, or we could get this merged and then I could do another PR. I have more new proposed changes as well, including some speed improvement.

Reply:
I see. So perhaps we can add a few inline comments about what's going on here and then yeah we can fix it in the downstream PR. 

So definitely the previous implementation was incorrect right?